src/main/java/org/folio/service/transactions/BatchTransactionService.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.service.transactions;
+
+import io.vertx.core.Future;
+import io.vertx.ext.web.handler.HttpException;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.jaxrs.model.Batch;
+import org.folio.rest.jaxrs.model.Transaction;
+
+import java.util.List;
+
+public class BatchTransactionService {
+
+  public Future<Void> processBatch(Batch batch, RequestContext requestContext) {
+    try {
+      sanityChecks(batch);
+    } catch (Exception ex) {
+      return Future.failedFuture(ex);
+    }
+    if (!batch.getTransactionsToCreate().isEmpty()) {
```

Comment:
CollectionUtils.isNotEmpty is more preferable here, because NPE is not possible and easier to read without !

Reply:
just small suggestion to use switch cases

Reply:
A switch would assume only one case is processed. Here we want to process all operations. I added more implementation to clarify.
Regarding `isNotEmpty()`: it would not hurt to use it, but I think `getTransactionsToCreate()` can never be null, and with my new implementation this is no longer needed.